huggingface_pipelines/text.py

```diff
@@ -0,0 +1,355 @@
+            batch_size=32,
+            device="cuda"
+        )
+        pipeline = HFTextToEmbeddingPipeline(config)
+
+    """
+
+    def __init__(self, config: TextToEmbeddingPipelineConfig):
+        """
+        Initialize the embedding-to-text pipeline.
+        Args:
+            config (EmbeddingToTextPipelineConfig): Configuration for the pipeline.
+        """
+        super().__init__(config)
+        logger.info("Initializing text to embedding model...")
+        self.config = config
+        self.t2vec_model = TextToEmbeddingModelPipeline(
+            encoder=self.config.encoder_model,
+            tokenizer=self.config.encoder_model,
+            device=self.config.device
+        )
+
+    def process_batch(self, batch: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Process a batch of embeddings and convert them to text.
+        Args:
+            batch (Dict[str, Any]): Input batch containing embeddings.
+        Returns:
+            Dict[str, List[str]]: Processed batch with decoded texts.
+        """
+        for column in self.config.columns:
+            if column in batch:
+                assert all(isinstance(item, list) for item in batch[column]), \
+                    f"Column {column} must contain only lists of sentences, not individual strings."
+                all_sentences = [sentence for item in batch[column]
+                                 for sentence in item]
+                all_embeddings = self.encode_texts(all_sentences)
+
+                sentence_embeddings = []
+                start_idx = 0
+                for item in batch[column]:
+                    end_idx = start_idx + len(item)
+                    sentence_embeddings.append(
+                        all_embeddings[start_idx:end_idx])
+                    start_idx = end_idx
+
+                batch[f"{column}_{self.config.output_column_suffix}"] = sentence_embeddings
+                logger.info(
+                    f" {column} column embeddings:  {batch[column][:5]}")
+            else:
+                logger.warning(f"Column {column} not found in batch.")
+        return batch
+
+    def encode_texts(self, texts: List[str]) -> np.ndarray:
+        """
+        Encode a list of texts into embeddings.
+        Args:
+            texts (List[str]): List of texts to encode.
+        Returns:
+            np.ndarray: Array of embeddings.
+        Raises:
+            Exception: If there's an error during encoding.
+        """
+        try:
+            embeddings = []
+            for i in range(0, len(texts), self.config.batch_size):
+                batch_texts = texts[i:i + self.config.batch_size]
+                batch_embeddings = self.t2vec_model.predict(
+                    batch_texts, source_lang=self.config.source_lang, batch_size=self.config.batch_size, max_seq_len=self.config.max_seq_len)
+
+                embeddings.extend(batch_embeddings)
+            return np.array(embeddings)
```

Comment:
Any reason why we must return a `np.ndarray` here ? This decision might have implications in the precision of the output embeddings